### PR TITLE
Search Functionality for Students

### LINKED ISSUE #56: Search Functionality
## Issue Description
As a student, I want to be able to search through all of the existing courses to find the right course. There should be some sort of search bar at the top of the dashboard which allows me to query the database display a list of courses on a new page. When I click on one of these courses, I should be taken to the course page.

## Tasks
- [x] Add search bar
- [x] Create results view
- [x] Add the necessary logic to query the database from the search string

### PR DESCRIPTION
**Pull request by brandon1024**
### Changes Proposed in this Pull Request:
- Moved duplicate code into single JS file
- Added search modal for student view, allowing students to search for a course by partial name (course name or course section)

### CODE REVIEW

**Comment by dsteen338** [comment]
Right away I noticed the modal is wrong, it's still the reuse seat grid one

**Comment by dsteen338** [comment]
Another small thing is if there is no course returned it would be nice if there was some display, not necessary

File: src/main/java/com/unb/tracker/model/Course.java [hunk 1]
```diff
@@ -27,7 +27,6 @@
     private List<Seat> seats;
 
     @ManyToOne(fetch = FetchType.EAGER)
-    @JsonIgnore
```
Comment (dsteen338) [inline src/main/java/com/unb/tracker/model/Course.java:29]: why?

**Review by dsteen338 — CHANGES_REQUESTED** [review]
Just a question on the code, and a couple of functional things I noticed:

* Another small thing is if there is no course returned it would be nice if there was some display, not necessary

* Right away I noticed the modal is wrong, it's still the reuse seat grid one

File: src/main/java/com/unb/tracker/model/Course.java (hunk 1 above)
Comment (tonerm43) [reply src/main/java/com/unb/tracker/model/Course.java:29]: We had to remove this annotation for the course instructor to be returned by the ajax call. We need the instructor name to be able to redirect to the right course

Comment (dsteen338) [reply src/main/java/com/unb/tracker/model/Course.java:29]: I believe @jacsmith21 added this, do you thing this will break anything else jacob?

Comment (jsmith) [reply src/main/java/com/unb/tracker/model/Course.java:29]: @dsteen338 It may. We'll just have to test to make sure everything works 👍

**Comment by dsteen338** [comment]
Testing, only real issue I found was entering special chars makes a bunch of toastr errors appear, but the whole system is unhappy with special chars so that's fine.

If we had time I would suggest clearing out the current results/ and or  adding a loading spinner when it's looking for courses, but I think this is fine for minimum viable. Approving.

**Review by dsteen338 — APPROVED** [review]

**Comment by jacsmith21** [comment]
image
This may not be a problem with this PR, but I'm getting the same course appear twice within the student view?

**Review by mwalz1 — APPROVED** [review]
Works well!

**Comment by jsmith** [comment]
Tested & merged!

**Pull request merged**